src/CoreWCF.Primitives/src/CoreWCF/Channels/MessageProperties.cs

```diff
@@ -471,7 +471,7 @@ internal void MergeProperties(MessageProperties properties)
             Encoder = properties.Encoder;
         }
 
-        internal void CopyProperties(KeyValuePair<string, object>[] array)
+        public void CopyProperties(KeyValuePair<string, object>[] array)
         {
```

Comment:
This code can be done inline at call site. The _disposed check isn't needed as this[key] does a disposal check, so you just need the loop, which can be simplified using foreach so can drop the array getter line. Basically inline:
```c#
foreach (var keypair in _properties)
{
    bufferedMessage.Properties[keypair.Key] = keypair.Value;
}
```

Reply:
This is addressed.